Vagrantfile

```diff
@@ -97,6 +98,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     vm_config.vm.host_name = "candlepin-el9.example.com"
     vm_config.vm.provision "shell", inline: "dnf update -y dnf ca-certificates"
     configure_ansible_provisioning(vm_config)
+
+    # Increase the partition size
+    vm_config.vm.provision "shell", inline: "growpart /dev/vda 1"
```

Comment:
We can accomplish this without extra dependencies with the following:

```    
vm_config.vm.disk :disk, size: "30GB", primary: true
vm_config.vm.provision "shell", inline: "echo '- +' | sfdisk --no-reread -N 1 /dev/vda && partprobe && resize2fs /dev/vda1"
```

This should also be the first operation we perform in the box. Configure -> correct -> provision

Reply:
As an example of what my config looked like for this test:

```
  config.vm.define("el9", autostart: false) do |vm_config|
    vm_config.vm.box = "centos.cloud/centos9s"
    vm_config.vm.box_url = "https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-Vagrant-9-latest.x86_64.vagrant-libvirt.box"
    vm_config.vm.host_name = "candlepin-el9.example.com"

    # Increase base disk size and resize partitions accordingly
    vm_config.vm.disk :disk, size: "30GB", primary: true
    vm_config.vm.provision "shell", inline: "echo '- +' | sfdisk --no-reread -N 1 /dev/vda && partprobe && resize2fs /dev/vda1"

    # Update DNF ca certs
    vm_config.vm.provision "shell", inline: "dnf update -y dnf ca-certificates"

    configure_ansible_provisioning(vm_config)
  end
```

Reply:
I reverted my changes and used your config that you provided, but when I run `df -H` the partition is still 12 GB inside of the Vagrant el9 VM.

```
(13:02:00) vagrant@candlepin-el9:~/devel/candlepin (jalbrech/m/CANDLEPIN-903) $ df -H
Filesystem                                                 Size  Used Avail Use% Mounted on
devtmpfs                                                   4.2M     0  4.2M   0% /dev
tmpfs                                                      2.1G   17k  2.1G   1% /dev/shm
tmpfs                                                      822M   12M  810M   2% /run
/dev/vda1                                                   12G  6.4G  4.6G  58% /
```

Reply:
I think the issue is that the box "virtual_size" metadata definition is overriding the vagrant native disk size definition (`vm_config.vm.disk :disk, size: "30GB", primary: true`). If you download the `CentOS-Stream-Vagrant-9-latest.x86_64.vagrant-libvirt.box` box and look at the metadata.json file it has `"virtual_size": 11`.

If I keep all of your config the same, but add the libvirt config `provider.machine_virtual_size = 20`, then the VM partition and filesystem is 20 GB as expected.

Reply:
Yeah, that's what I was mentioning above by needing both still. The disk size configuration is still new, so may not work universally/everywhere yet.

Also make sure the sizes line up so we don't get different results depending on which is used.

Reply:
My bad, I misread that portion of your comment! Updated and I used 30 GB for both configurations.